File: 3498.reverse-degree-of-a-string.ts
import { expect } from "bun:test";
/*
 * @lc app=leetcode id=3498 lang=typescript
 *
 * [3498] Reverse Degree of a String
 */

// @lc code=start
function reverseDegree(s: string): number {
  const BASE = 123;
  let ans = 0;

  for (let i = 0, len = s.length; i < len; i++) {
    ans += (BASE - s.codePointAt(i)!) * (i + 1);
  }
  return ans;
}
// @lc code=end

expect(reverseDegree("abc")).toBe(148);
expect(reverseDegree("zaza")).toBe(160);

/**
 * Accepted
 * 933/933 cases passed (0 ms)
 * Your runtime beats 100 % of typescript submissions
 * Your memory usage beats 73.21 % of typescript submissions (57.8 MB)
 */

File: src/c/3498.reverse-degree-of-a-string.c
#include <stdio.h>

/*
 * @lc app=leetcode id=3498 lang=c
 *
 * [3498] Reverse Degree of a String
 */

// @lc code=start
int reverseDegree(char *s)
{
    int ans = 0;

    for (int i = 0; s[i] != '\0'; i++)
    {
        ans += ((26 - (s[i] - 97)) * (i + 1));
    }
    return ans;
}
// @lc code=end

int main(void)
{
    printf("answer 1: %d", reverseDegree("abc") == 148);
    printf("answer 2: %d", reverseDegree("zaza") == 160);

    return 0;
}

/*
    Accepted
    933/933 cases passed (0 ms)
    Your runtime beats 100 % of c submissions
    Your memory usage beats 64.65 % of c submissions (9.2 MB)
*/